crates/reward/src/lib.rs

```diff
@@ -161,26 +164,24 @@ impl<T: Config> Pallet<T> {
             .ok_or(Error::<T>::TryIntoIntError)?)
     }
 
-    pub fn deposit_stake(
-        currency_id: T::CurrencyId,
-        account_id: &DefaultVaultId<T>,
-        amount: SignedFixedPoint<T>,
-    ) -> Result<(), DispatchError> {
-        checked_add_mut!(Stake<T>, currency_id, account_id, &amount);
-        checked_add_mut!(TotalStake<T>, currency_id, &amount);
-
-        <RewardTally<T>>::mutate(currency_id, account_id, |reward_tally| {
-            let reward_per_token = Self::reward_per_token(currency_id);
-            let reward_per_token_mul_amount = reward_per_token
-                .checked_mul(&amount)
-                .ok_or(Error::<T>::ArithmeticOverflow)?;
-            *reward_tally = reward_tally
-                .checked_add(&reward_per_token_mul_amount)
-                .ok_or(Error::<T>::ArithmeticOverflow)?;
-            Ok::<_, Error<T>>(())
-        })?;
+    pub fn deposit_stake(vault_id: &DefaultVaultId<T>, amount: SignedFixedPoint<T>) -> Result<(), DispatchError> {
+        for currency_id in [vault_id.wrapped_currency(), T::GetGovernanceTokenCurrency::get()] {
+            checked_add_mut!(Stake<T>, currency_id, vault_id, &amount);
+            checked_add_mut!(TotalStake<T>, currency_id, &amount);
```

Comment:
I think stake should be stored per `VaultId`, please see my changes here: https://github.com/interlay/interbtc/pull/286/files#diff-63acba1a1c0ed1a214390cea807f4b5f3e17506e74aa37617bae05d6c1950f45R178-R197

Reply:
The `staking` pallet was already changed here: https://github.com/interlay/interbtc/pull/273

Reply:
I purposefully left the stake to be stored per `(currencyId, vaultId)` tuple, since in the future we'll have to use a different value for the stake of the KINT - vaults with different backing currencies will need to share the KINT reward pool.

Reply:
I think I would prefer to have one storage item for now and split out when necessary - as per the recent discussion we should limit scope creep relative to what _we may_ need in the future.

Reply:
I agree in general, but I feel this is a bit of a special case, because this whole feature of generalizing over the wrapped currency is a case of anticipation of a possible future need, and also, this is a case where you want to _remove_ an existing feature, which goes a step further than _not adding_ a new feature. I guess this difference of opinion on how far to take the preparation for multi-wrapped support is part of the reason why we ended up with the two PRs implementing the reward change.

To be able to merge this pr asap (and because I do indeed see the value of not complicating code for anticipated uncertain future changes), I'll go and make the change.